### PR TITLE
Ariakit: Update to v0.4.10

### PR DESCRIPTION
**Pull request by tyxla**
## What?
Updates `@ariakit/react` to the latest `v0.4.10`.

I'm planning to use the new APIs in subsequent PRs.

## Why?
Mostly to consider a few fixes and the new `useStoreState` API that helps with React Compiler compatibility. Will be done in another PR. 

## How?
Just updating the dependency and resolving test failures where necessary.

## Testing Instructions
* Smoke all Ariakit components (including `Composite`, `CustomSelectControl`, `DropdownMenuV2`, `RadioGroup`, `TabPanel`, `Tabs`, `ToggleGroupControl`, `Tooltip`, `Toolbar`)
* Verify all checks are green.

### Testing Instructions for Keyboard
Verify all the above components are navigable with keyboard navigation like before.

### CODE REVIEW

**Review by ciampo — COMMENTED** [review]
Things are looking good after an initial smoke test.

Should we migrate `store.useState( stateProp )`  to `useStoreState(store, stateProp)` (new 0.4.9 feature), or is it unnecessary / premature?

**Comment by tyxla** [comment]
> Should we migrate `store.useState( stateProp )` to `useStoreState(store, stateProp)` (new 0.4.9 feature), or is it unnecessary / premature?

I'm planning to do it in a follow-up PR.

File: package-lock.json [hunk 1]
```diff
@@ -67558,8 +67558,7 @@
 					"integrity": "sha512-GUy/3ZY4kW1KdYHtMZRrRlj5FYbZTAyHlimxHI7Zs0xsC+kAzIf8lopnf67Y9IYLgEMr37KosIV7kwpkJpNs5Q=="
 				},
 				"@ariakit/react": {
-					"version": "0.4.7",
-					"resolved": "https://registry.npmjs.org/@ariakit/react/-/react-0.4.7.tgz",
+					"version": "https://registry.npmjs.org/@ariakit/react/-/react-0.4.7.tgz",
```
Comment (ciampo) [inline package-lock.json]: This new version of `package-lock.json` still points at a `0.4.7` file, is this expected?

Comment (tyxla) [reply package-lock.json]: Nope, this isn't right, I'll take a look

Comment (tyxla) [reply package-lock.json]: Should be resolved by b937a27cb580d5eb63b462a97adbe8fa6503ddf6. 

There was a dependency duplication going on, and I addressed it.

**Comment by tyxla** [comment]
Now that the dependency was correctly updated, we'll see some failures related with the `v0.4.8` tab changes, I'll take a look at them.

**Comment by ciampo** [comment]
Adding `accessibleWhenDisabled` to `Tabs.Tab` seems to solve the issue on my local machine — although I thought that ariakit's `Tab` [already defaulted it to `true`](https://ariakit.org/reference/tab#accessiblewhendisabled) 🤔

**Comment by tyxla** [comment]
> Adding `accessibleWhenDisabled` to `Tabs.Tab` seems to solve the issue on my local machine — although I thought that ariakit's `Tab` [already defaulted it to `true`](https://ariakit.org/reference/tab#accessiblewhendisabled) 🤔

Yup, seems like it might be coming from here: https://github.com/ariakit/ariakit/pull/4018

**Comment by ciampo** [comment]
> > Adding `accessibleWhenDisabled` to `Tabs.Tab` seems to solve the issue on my local machine — although I thought that ariakit's `Tab` [already defaulted it to `true`](https://ariakit.org/reference/tab#accessiblewhendisabled) 🤔
> 
> Yup, seems like it might be coming from here: [ariakit/ariakit#4018](https://github.com/ariakit/ariakit/pull/4018)

I'm surprised that the changed you linked affected us, since we never set `accessibleWhenDisabled` to `false` explicitly, and we instead relied on the fact that `accessibleWhenDisabled` has an implicit value of `true`.

Basically, it's unexpected to me that we have to set it explicitly, since I thought it already always defaulted to `true`.

(cc @diegohaz in case he can clarify why this is happening)

File: packages/components/src/tab-panel/index.tsx [hunk 2]
```diff
@@ -220,6 +220,7 @@ const UnforwardedTabPanel = (
 									showTooltip={ !! tab.icon }
 								/>
 							}
+							accessibleWhenDisabled
```
Comment (ciampo) [inline packages/components/src/tab-panel/index.tsx]: We may to have to do the same for the `Tab` component

```diff
diff --git a/packages/components/src/tabs/tab.tsx b/packages/components/src/tabs/tab.tsx
index e1aa85c636..8ab67e1ca9 100644
--- a/packages/components/src/tabs/tab.tsx
+++ b/packages/components/src/tabs/tab.tsx
@@ -26,6 +26,7 @@ export const Tab = forwardRef<
 	const instancedTabId = `${ instanceId }-${ tabId }`;
 	return (
 		<StyledTab
+			accessibleWhenDisabled
 			ref={ ref }
 			store={ store }
 			id={ instancedTabId }

```

**Comment by tyxla** [comment]
> > > Adding `accessibleWhenDisabled` to `Tabs.Tab` seems to solve the issue on my local machine — although I thought that ariakit's `Tab` [already defaulted it to `true`](https://ariakit.org/reference/tab#accessiblewhendisabled) 🤔
> > 
> > 
> > Yup, seems like it might be coming from here: [ariakit/ariakit#4018](https://github.com/ariakit/ariakit/pull/4018)
> 
> I'm surprised that the changed you linked affected us, since we never set `accessibleWhenDisabled` to `false` explicitly, and we instead relied on the fact that `accessibleWhenDisabled` has an implicit value of `true`.
> 
> Basically, it's unexpected to me that we have to set it explicitly, since I thought it already always defaulted to `true`.
> 
> (cc @diegohaz in case he can clarify why this is happening)

Agreed it's surprising - it is also documented to be `true` by default: https://ariakit.org/reference/tab#accessiblewhendisabled

File: packages/components/src/tab-panel/index.tsx (hunk 2 above)
Comment (tyxla) [reply packages/components/src/tab-panel/index.tsx]: Done in 312804105575044a83d6086268ec50cee4da26bb.

**Comment by tyxla** [comment]
Note that ariakit/ariakit#4018 seems to be specifically removing the default `true` of `accessibleWhenDisabled`:

https://github.com/ariakit/ariakit/pull/4018/files#diff-1b1635ca34878fe6ca473aeecac2f52b9aaa31af0ed287d9c2b9ec14c5ebcf64L40

**Comment by diegohaz** [comment]
> Note that [ariakit/ariakit#4018](https://github.com/ariakit/ariakit/pull/4018) seems to be specifically removing the default `true` of `accessibleWhenDisabled`:
> 
> https://github.com/ariakit/ariakit/pull/4018/files#diff-1b1635ca34878fe6ca473aeecac2f52b9aaa31af0ed287d9c2b9ec14c5ebcf64L40

Yes! 🤦🏽‍♂️

That was unintentional.

**Comment by tyxla** [comment]
Here's a quick PR to reverse it with as little changes as needed: https://github.com/ariakit/ariakit/pull/4056 cc @diegohaz

**Comment by tyxla** [comment]
Currently blocked by https://github.com/ariakit/ariakit/pull/4056

Will resume when there's a minor version that fixes the default of `accessibleWhenDisabled`.

**Comment by tyxla** [comment]
OK, I've updated to the latest version with the `accessibleWhenDisabled` fix ([v0.4.10](https://github.com/ariakit/ariakit/pull/4054)) and reverted the `accessibleWhenDisabled` patches. 

Should be ready for another look!

File: packages/components/CHANGELOG.md [hunk 3]
```diff
@@ -68,10 +68,8 @@
 -   `Composite` v2: add `Hover` and `Typeahead` subcomponents ([#64399](https://github.com/WordPress/gutenberg/pull/64399)).
 -   `Composite` v2: add focus-related props to `Composite`and`Composite.Item` subcomponents ([#64450](https://github.com/WordPress/gutenberg/pull/64450)).
 -   `Composite` v2: add `Context` subcomponent ([#64493](https://github.com/WordPress/gutenberg/pull/64493)).
-
-### Internal
-
 -   `CustomSelectControl`: Improve type inferring ([#64412](https://github.com/WordPress/gutenberg/pull/64412)).
+-   Update `ariakit` to version `0.4.9` ([#64637](https://github.com/WordPress/gutenberg/pull/64637)).
```
Comment (ciampo) [inline packages/components/CHANGELOG.md]:
```suggestion
-   Update `ariakit` to version `0.4.10` ([#64637](https://github.com/WordPress/gutenberg/pull/64637)).
```

**Review by ciampo — APPROVED** [review]
🚀

**Pull request merged**